src/System.Diagnostics.Process/src/System/Diagnostics/Process.OSX.cs

```diff
@@ -86,8 +88,31 @@ private void GetWorkingSetLimits(out IntPtr minWorkingSet, out IntPtr maxWorking
         /// <param name="resultingMax">The resulting maximum working set limit after any changes applied.</param>
         private void SetWorkingSetLimitsCore(IntPtr? newMin, IntPtr? newMax, out IntPtr resultingMin, out IntPtr resultingMax)
         {
-            // TODO: Implement this
-            throw NotImplemented.ByDesign;
+            // There isn't a way to set the minimum working set, so throw an exception here
+            if (newMin.HasValue)
+            {
+                throw new PlatformNotSupportedException(SR.MinimumWorkingSetNotSupported);
+            }
+
+            // The minimum resident set will always be 0, default the resulting max to 0 until we set it (to make the compiler happy)
+            resultingMin = IntPtr.Zero;
+            resultingMax = IntPtr.Zero;
+
+            // The default hard upper limit is absurdly high (over 9000PB) so just change the soft limit...especially since
+            // if you aren't root and move the upper limit down, you need root to move it back up
+            if (newMax.HasValue)
+            {
+                Interop.libc.rlimit limits = new Interop.libc.rlimit() { rlim_cur = (ulong)newMax.Value.ToInt64() };
+                int result = Interop.libc.setrlimit(Interop.libc.RLIMIT_Resources.RLIMIT_RSS, ref limits);
```

Comment:
Good catch. I couldn't find a way to set this on other processes so I'll throw a PlatformNotSupported exception in this scenario